> But, unfortunately, test still fail with the same `MODULE_NOT_FOUND @prisma/engines-version` error. Not sure how to fix it.
> 
> The package itself is working correctly though.

The package itself is broken as well, as far as I can see:

```
$ ./result/bin/prisma --version
node:internal/modules/cjs/loader:1252
  throw err;
  ^

Error: Cannot find module '@prisma/engines-version'
Require stack:
- /nix/store/ag0b74n1iz6hwf52r9mbn17xkclq6qb0-prisma-6.3.0/lib/prisma/packages/engines/dist/index.js
- /nix/store/ag0b74n1iz6hwf52r9mbn17xkclq6qb0-prisma-6.3.0/lib/prisma/packages/cli/build/index.js
    at Function._resolveFilename (node:internal/modules/cjs/loader:1249:15)
    at Function._load (node:internal/modules/cjs/loader:1075:27)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:219:24)
    at Module.require (node:internal/modules/cjs/loader:1340:12)
    at require (node:internal/modules/helpers:138:16)
    at Object.<anonymous> (/nix/store/ag0b74n1iz6hwf52r9mbn17xkclq6qb0-prisma-6.3.0/lib/prisma/packages/engines/dist/index.js:41:30)
    at Module._compile (node:internal/modules/cjs/loader:1565:14)
    at Object..js (node:internal/modules/cjs/loader:1708:10)
    at Module.load (node:internal/modules/cjs/loader:1318:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/nix/store/ag0b74n1iz6hwf52r9mbn17xkclq6qb0-prisma-6.3.0/lib/prisma/packages/engines/dist/index.js',
    '/nix/store/ag0b74n1iz6hwf52r9mbn17xkclq6qb0-prisma-6.3.0/lib/prisma/packages/cli/build/index.js'
  ]
}
```

--- a/nixos/doc/manual/release-notes/rl-2505.section.md
+++ b/nixos/doc/manual/release-notes/rl-2505.section.md
@@ -337,7 +337,7 @@
   add `vimPlugins.notmuch-vim` to your (Neo)vim configuration if you want the
   vim plugin.
 
-- `prisma` and `prisma-engines` have been updated to version 6.0.1, which
+- `prisma` and `prisma-engines` have been updated to version 6.3.0, which
   introduces several breaking changes. See the
   [Prisma ORM upgrade guide](https://www.prisma.io/docs/orm/more/upgrade-guides/upgrading-versions/upgrading-to-prisma-6)
   for more information.

--- a/pkgs/by-name/pr/prisma/package.nix
+++ b/pkgs/by-name/pr/prisma/package.nix
@@ -3,7 +3,7 @@
   fetchFromGitHub,
   stdenv,
   nodejs,
-  pnpm_8,
+  pnpm_9,
   prisma-engines,
   jq,
   makeWrapper,
@@ -13,26 +13,26 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "prisma";
-  version = "6.0.1";
+  version = "6.3.0";
 
   src = fetchFromGitHub {
     owner = "prisma";
     repo = "prisma";
     rev = finalAttrs.version;
-    hash = "sha256-mwGFuJLry2WvwLclRw+ulMVgp8tfZbhzrdgKjQ4D7LE=";
+    hash = "sha256-Buu+E0xxjcrPOyEHkQTp7IVS9kymmR1PTegeOXxb2PA=";
   };
 
   nativeBuildInputs = [
     nodejs
-    pnpm_8.configHook
+    pnpm_9.configHook
     jq
     makeWrapper
     moreutils
   ];
 
-  pnpmDeps = pnpm_8.fetchDeps {
+  pnpmDeps = pnpm_9.fetchDeps {
     inherit (finalAttrs) pname version src;
-    hash = "sha256-fOg32w/fQkyn8HBMffUKob7XzOQLtsB642pDdEz/y2E=";
+    hash = "sha256-rAEUkk3uWVuUDrSRz6d2Ewr3vi4rzYmO0yLTCl21qZ4=";
   };
 
   patchPhase = ''
@@ -65,7 +65,7 @@ stdenv.mkDerivation (finalAttrs: {
 
     # Remove unnecessary external dependencies
     rm -rf node_modules
-    pnpm install --offline --ignore-scripts --frozen-lockfile --prod --filter ./packages/cli
+    pnpm install --offline --ignore-scripts --frozen-lockfile --prod
     cp -r node_modules $out/lib/prisma
 
     # Only install cli and its workspace dependencies

--- a/pkgs/development/tools/database/prisma-engines/default.nix
+++ b/pkgs/development/tools/database/prisma-engines/default.nix
@@ -13,27 +13,24 @@
 # function correctly.
 rustPlatform.buildRustPackage rec {
   pname = "prisma-engines";
-  version = "6.0.1";
+  version = "6.3.0";
 
   src = fetchFromGitHub {
     owner = "prisma";
     repo = "prisma-engines";
     rev = version;
-    hash = "sha256-wrhOUUF3N0JvDxcqypL7q+B7SwAfLq9M0Zqs86jnB/M=";
+    hash = "sha256-gQLDskabTaNk19BJi9Kv4TiEfVck2QZ7xdhopt5KH6M=";
   };
 
   useFetchCargoVendor = true;
-  cargoHash = "sha256-l5b7TcVgnTdUsUncvfJYBcmh0xhF3fMfoCx0cyjpJ8Y=";
+  cargoHash = "sha256-GLOGivOH8psE5/M5kYakh9Cab4Xe5Q8isY1c6YDyAB8=";
 
   # Use system openssl.
   OPENSSL_NO_VENDOR = 1;
 
   nativeBuildInputs = [ pkg-config ];
 
-  buildInputs = [
-    openssl
-    protobuf
-  ];
+  buildInputs = [ openssl ];
 
   preBuild = ''
     export OPENSSL_DIR=${lib.getDev openssl}

